internal/adapters/runtime/kubernetes/scheduler.go

```diff
@@ -219,6 +219,10 @@ func (k *kubernetes) MitigateDisruption(
 		return nil
 	}
 
+	pdb, err = k.clientSet.PolicyV1().PodDisruptionBudgets(scheduler.Name).Get(ctx, scheduler.Name, metav1.GetOptions{})
+	if err != nil || pdb == nil {
+		return errors.NewErrUnexpected("non recoverable error when getting PDB for scheduler '%s': %s", scheduler.Name, err)
+	}
```

Comment:
In theory, the PDB object can be changed in the runtime in the meantime after we getting the initial state/ask for creation (beginning of the function) and before updating the value.  This should never happen in production because there is only one agent setting this PDB in the namespace and it's Maestro worker. However, on tests we were seeing intermittent failures running parallel cases, hence why I'm adding this to be safer before updating the object

Reply:
This is the kind of error thrown in that scenario: https://github.com/topfreegames/maestro/actions/runs/9422115862/job/25957670592

Reply:
Nice, thanks for the clarification. It is kinda counterintuitive just by reading the code, perhaps add a small context as a comment?